src/js/ProductDetails.mjs

```diff
@@ -1,68 +1,46 @@
-import { setLocalStorage } from "./utils.mjs"
+import { getLocalStorage, setLocalStorage } from "./utils.mjs";
 
 function productDetailsTemplate(product) {
-    return `<section class="product-detail">
+  return `<section class="product-detail">
         <h3>${product.Brand.Name}</h3>
         <h2 class="divider">${product.NameWithoutBrand}</h2>
         <img class="divider" src="${product.Image}" alt="${product.NameWithoutBrand}"></img>
         <p class="product-card__price">$${product.FinalPrice}</p>
         <p class="product__color">${product.Colors[0].ColorName}</p>
-        <p class="product__description">${product.DescriptionHTMLSimple}</p>
+        <p class="product__description">${product.DescriptionHtmlSimple}</p>
         <div class="product-detail__add">
             <button id="addProductToCart" data-id="${product.Id}">Add to Cart</button>
         </div>
     </section>`;
 }
 
-
 export default class ProductDetails {
-
-    constructor(productId, dataSource){
-        this.productId = productId;
-        this.product = {};
-        this.dataSource = dataSource;
-    }
-    
-    async init() {
-        this.product = await this.dataSource.findProductById(this.productId);
-        this.renderProductDetails("main");
-          // add listener to Add to Cart button
-        document
-            .getElementById("addProductToCart") //added 'Product' - addToCart original
-            .addEventListener("click", addToCartHandler);
-    }
-    
-    addProductToCart() {
-        setLocalStorage("so-cart", this.product);
-    }
-    
-    renderProductDetails(selector) {
-        const element = document.querySelector(selector);
-        element.insertAdjacentHTML(
-            "afterBegin",
-            productDetailsTemplate(this.product)
-        );
-    }
+  constructor(productId, dataSource) {
+    this.productId = productId;
+    this.product = {};
+    this.dataSource = dataSource;
+  }
+
+  async init() {
+    this.product = await this.dataSource.findProductById(this.productId);
+    this.renderProductDetails("main");
+    // add listener to Add to Cart button
+    document
+      .getElementById("addProductToCart") //added 'Product' - addToCart original
+      .addEventListener("click", () => this.addToCartHandler());
```

Comment:
In order for `this.product` to work in `addToCartHandler` you need to use a fat arrow function like this or:

```
.addEventListener("click", this.addToCartHandler.bind(this));
```

Otherwise Javascript will assume `this` is the button and that won't have a product.